src/units/unit.hpp

```diff
@@ -1054,6 +1054,8 @@ class unit : public std::enable_shared_from_this<unit>
 private:
 	bool resistance_filter_matches(const config& cfg, bool attacker, const std::string& damage_name, int res) const;
 
+	int resistance_abilitie(unit_ability_list resistance_abilities, const std::string& damage_name, bool attacker, int res) const;
```

Comment:
I feel this needs a better name, but don't have a better suggestion yet. Documentation would help.